File: src/common/js/ArmstrongNumbers.js
function isArmstrong(number) {
    const digitCount = number.length;

    let sum = 0;
    for (let digitAsString of number) {
        let digit = parseInt(digitAsString, 10);
        sum += digit ** digitCount;
    }
    return sum == number;
}

function getNthArmstrongNumber(n) {
    let count = 0;
    for(let i = 1; i <= Number.MAX_VALUE; i++) {
        if(isArmstrong(pari)) {
            count++;

            if(count == n) {
                return i;
            }
        }
    }
}

console.log(isArmstrong('153'))
console.log(isArmstrong('120'))
console.log(isArmstrong(12))

File: src/common/java/ArmstrongNumbers.java
package common.java;

/**
 * https://www.geeksforgeeks.org/program-for-armstrong-numbers/
 */
public class ArmstrongNumbers {
    public static void main(String[] args) {
        System.out.println(isArmstrong(153));
        System.out.println(isArmstrong(153));
        System.out.println(getNthArmstrong(12));
    }

    private static boolean isArmstrong(int number) {
        String numberAsString = String.valueOf(number);
        int size = numberAsString.length();

        int sum = 0;
        for(char ch : numberAsString.toCharArray()) {
            int characterAsInt = Integer.parseInt(ch+"");
            sum += Math.pow(characterAsInt, size);
        }
        return sum == number;
    }

    private static int getNthArmstrong(int n) {
        int count = 0;
        for(int i = 1; i < Integer.MAX_VALUE; i++) {
            if(isArmstrong(i) == true) {
                count++;
                if(count == n) { return i; }
            }
        }
        return count;
    }
}
